File: backup_java/dialogs/DialogBuyFromMerchant.java
package it.paranoidsquirrels.idleguildmaster.ui.dialogs;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.viewbinding.ViewBinding;
import it.paranoidsquirrels.idleguildmaster.MainActivity;
import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.UIUtils;
import it.paranoidsquirrels.idleguildmaster.databinding.DialogBuyFromMerchantBinding;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.MerchantOffer;
import java.util.function.BooleanSupplier;

/* JADX INFO: loaded from: classes3.dex */
public class DialogBuyFromMerchant extends CustomDialog {
    public DialogBuyFromMerchantBinding binding;
    public BooleanSupplier callback;
    public MerchantOffer offer;

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding getBinding() {
        return this.binding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void setBinding(ViewBinding viewBinding) {
        this.binding = (DialogBuyFromMerchantBinding) viewBinding;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected String getTitle() {
        return getString(R.string.merchant_dialog_confirm_buy_title);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void setLayout() {
        getDialog().getWindow().setLayout((int) (((double) getResources().getDisplayMetrics().widthPixels) * 0.9d), -2);
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected ViewBinding inflate(LayoutInflater layoutInflater, ViewGroup viewGroup, boolean z) {
        DialogBuyFromMerchantBinding dialogBuyFromMerchantBindingInflate = DialogBuyFromMerchantBinding.inflate(layoutInflater, viewGroup, z);
        this.binding = dialogBuyFromMerchantBindingInflate;
        return dialogBuyFromMerchantBindingInflate;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void initialize(Bundle bundle) {
        this.binding.confirmText.setText(this.offer.getItem().getStack() > 1 ? String.format(getString(R.string.merchant_dialog_confirm_buy_multiple_body), Integer.valueOf(this.offer.getItem().getStack()), getString(this.offer.getItem().getIdName())) : String.format(getString(R.string.merchant_dialog_confirm_buy_single_body), getString(this.offer.getItem().getIdName())));
        int i = 8;
        this.binding.containerMoney.setVisibility(this.offer.isGems() ? 8 : 0);
        this.binding.containerGems.setVisibility(this.offer.isGems() ? 0 : 8);
        TextView textView = this.binding.shop;
        // Temporary disabled in UI per configuration (implemented but hidden)
        textView.setVisibility(8);
        if (this.offer.isGems()) {
            this.binding.amountGems.setText(String.valueOf(this.offer.getPrice()));
        } else {
            UIUtils.populateMoneyContainer(this.binding.amountMoney, this.offer.getPrice(), true);
        }
        this.binding.error.setTextColor(getResources().getColor(UIUtils.getFailureColor(), getContext().getTheme()));
    }

    @Override // it.paranoidsquirrels.idleguildmaster.ui.dialogs.CustomDialog
    protected void attachListeners() {
        this.binding.cancel.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBuyFromMerchant$$ExternalSyntheticLambda0
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogBuyFromMerchant.this.m226xa03305b7(view);
            }
        });
        this.binding.priceContainer.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBuyFromMerchant$$ExternalSyntheticLambda1
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogBuyFromMerchant.this.m227x835eb8f8(view);
            }
        });
        this.binding.shop.setOnClickListener(new View.OnClickListener() { // from class: it.paranoidsquirrels.idleguildmaster.ui.dialogs.DialogBuyFromMerchant$$ExternalSyntheticLambda2
            @Override // android.view.View.OnClickListener
            public final void onClick(View view) {
                DialogBuyFromMerchant.this.m228x668a6c39(view);
            }
        });
    }

    /* JADX INFO: renamed from: lambda$attachListeners$0$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogBuyFromMerchant, reason: not valid java name */
    /* synthetic */ void m226xa03305b7(View view) {
        dismiss();
    }

    /* JADX INFO: renamed from: lambda$attachListeners$1$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogBuyFromMerchant, reason: not valid java name */
    /* synthetic */ void m227x835eb8f8(View view) {
        if (this.callback.getAsBoolean()) {
            dismiss();
        }
    }

    /* JADX INFO: renamed from: lambda$attachListeners$2$it-paranoidsquirrels-idleguildmaster-ui-dialogs-DialogBuyFromMerchant, reason: not valid java name */
    /* synthetic */ void m228x668a6c39(View view) {
        if (MainActivity.shownDialogShop != null) {
            return;
        }
        new DialogShop().show(getParentFragmentManager(), "shop");
    }

    public void writeError(String str) {
        this.binding.error.setText(str);
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStart() {
        super.onStart();
        MainActivity.shownDialogBuyFromMerchant = this;
    }

    @Override // androidx.fragment.app.DialogFragment, androidx.fragment.app.Fragment
    public void onStop() {
        MainActivity.shownDialogBuyFromMerchant = null;
        super.onStop();
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/ui/dialogs/DialogBuyFromMerchant.kt
package it.paranoidsquirrels.idleguildmaster.ui.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import it.paranoidsquirrels.idleguildmaster.MainActivity
import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.UIUtils
import it.paranoidsquirrels.idleguildmaster.databinding.DialogBuyFromMerchantBinding
import it.paranoidsquirrels.idleguildmaster.storage.data.items.MerchantOffer
import java.util.function.BooleanSupplier

class DialogBuyFromMerchant : CustomDialog() {
    @JvmField
    var binding: DialogBuyFromMerchantBinding? = null
    @JvmField
    var callback: BooleanSupplier? = null
    @JvmField
    var offer: MerchantOffer? = null

    override fun getBinding(): ViewBinding = binding!!

    override fun setBinding(viewBinding: ViewBinding) {
        binding = viewBinding as DialogBuyFromMerchantBinding
    }

    override fun getTitle(): String = getString(R.string.merchant_dialog_confirm_buy_title)

    override fun setLayout() {
        dialog?.window?.setLayout((resources.displayMetrics.widthPixels * 0.9).toInt(), -2)
    }

    override fun inflate(inflater: LayoutInflater, container: ViewGroup?, attachToRoot: Boolean): ViewBinding {
        val b = DialogBuyFromMerchantBinding.inflate(inflater, container, attachToRoot)
        binding = b
        return b
    }

    override fun initialize(arguments: Bundle?) {
        val off = offer ?: return
        val b = binding ?: return
        val item = off.item ?: return
        b.confirmText.text = if (item.getStack() > 1) {
            String.format(getString(R.string.merchant_dialog_confirm_buy_multiple_body), item.getStack(), getString(item.getIdName()))
        } else {
            String.format(getString(R.string.merchant_dialog_confirm_buy_single_body), getString(item.getIdName()))
        }
        b.containerMoney.visibility = if (off.isGems) 8 else 0
        b.containerGems.visibility = if (off.isGems) 0 else 8
        b.shop.visibility = 8
        if (off.isGems) {
            b.amountGems.text = off.price.toString()
        } else {
            UIUtils.populateMoneyContainer(b.amountMoney, off.price, true)
        }
        context?.let { ctx ->
            b.error.setTextColor(resources.getColor(UIUtils.getFailureColor(), ctx.theme))
        }
    }

    override fun attachListeners() {
        val b = binding ?: return
        b.cancel.setOnClickListener {
            dismiss()
        }
        b.priceContainer.setOnClickListener {
            if (callback?.asBoolean == true) {
                dismiss()
            }
        }
        b.shop.setOnClickListener {
            if (MainActivity.shownDialogShop == null) {
                DialogShop().show(parentFragmentManager, "shop")
            }
        }
    }

    fun writeError(str: String) {
        binding?.error?.text = str
    }

    override fun onStart() {
        super.onStart()
        MainActivity.shownDialogBuyFromMerchant = this
    }

    override fun onStop() {
        MainActivity.shownDialogBuyFromMerchant = null
        super.onStop()
    }
}
